### PR TITLE
[RFC] vim-patch:7.4.1568,7.4.1571,7.4.1728

### PR DESCRIPTION
**Pull request by jamessan**
patch 7.4.1568
Problem:    Using CTRL-] in help on option in parentheses doesn't work.
Solution:   Skip the "(" in "('". (Hirohito Higashi)

https://github.com/vim/vim/commit/00f9e0dbbd3472db217d56639fad9346b9eb3b82

patch 7.4.1571
Problem:    No test for ":help"
Solution:   Add a test for what 7.4.1568 fixed. (Higashi Higashi)

https://github.com/vim/vim/commit/8e15ffcde757ffc6cfe8b5e384948b3278e9af33

patch 7.4.1728
Problem:    The help for functions require a space after the "(".
Solution:   Make CTRL-] on a function name ignore the arguments. (Hirohito
            Higashi)

https://github.com/vim/vim/commit/81edd171a9465cf99cede4fa4a7b7bca3d538b0f

### CODE REVIEW

**Comment by jamessan** [comment]
This supersedes #4659.

File: Makefile [hunk 1]
```diff
@@ -85,9 +85,12 @@ endif
 	mkdir -p build
 	touch $@
 
-oldtest: | nvim
+oldtest: | nvim tags
```
Comment (jamessan) [inline Makefile]: I debated between this or generating minimal help and tags files in the test itself.  I had originally (when first picking up brcolow's work) done the latter but then revised the test to just set 'runtimepath'.

**Comment by justinmk** [comment]
Do we need to run `:helptags` to force the tags to build?

**Comment by jamessan** [comment]
Why does `.ci/common/test.sh` directly run various command lines instead of using the targets defined in Makefile?  Shouldn't the logic be contained in Makefile instead of duplicated in the CI scripts?

File: Makefile (hunk 1 above)
Comment (justinmk) [inline Makefile]: Hard-coding a help tags fragment in the test and saving it with `helpers.write_file()` would be more than fine.

**Comment by justinmk** [comment]
> Why does .ci/common/test.sh directly run various command lines instead of using the targets defined in Makefile? 

Just legacy naivete at the time. Definitely in favor of avoiding that if possible.

> Shouldn't the logic be contained in Makefile instead of duplicated in the CI scripts?

Yes, been nudging PRs in that direction.

File: Makefile (hunk 1 above)
Comment (jamessan) [inline Makefile]: Well, this is a legacy test, so helpers isn't available.  If the CI scripts weren't duplicating Makefile's knowledge, this should work.

**Pull request merged**